coding-guides/starboard.md

```diff
@@ -0,0 +1,281 @@
+# Starboard
+
+This is a long awaited feature, requested by many people.
+
+Let's begin by talking about what a starboard is. This is an example taken from the Discord.js Official Server.
+
+![Starboard](/assets/starboard.png)
+
+> Before we start, there is one thing you need to now. This code is compatible with only Guidebot Class. This is due to a restriction with passing `client` to the messageReactionAdd and messageReactionRemove events.
+
+So, let's begin!
+
+```js
+module.exports = class {
+  constructor(client) {
+    this.client = client;
+
+    this.starEmbed = async (color, description, author, authorURL, timestamp, footer, image) => {
+      const embed = { 
+        'color': color, 
+        'description': description, 
+        'author': { 
+          'name': author,
+          'icon_url': authorURL
+        },
+        'image': { 
+          'url': image 
+        }, 
+        'timestamp': timestamp, 
+        'footer': { 
+          'text': footer 
+        } 
+      };
+      return embed; 
+    };
+
+    this.extension = async (reaction, attachment) => {
+      const imageLink = attachment.split('.');
+      const typeOfImage = imageLink[imageLink.length - 1];
+      const image = /(jpg|jpeg|png|gif)/gi.exec(typeOfImage);
+      if (!image) return '';
+      return attachment;
+    };
+  }
+}
+```
+
+Alright, let's break this down. 
+
+```js
+this.starEmbed = async (color, description, author, authorURL, timestamp, footer, image) => {
+  const embed = { 
+    'color': color, 
+    'description': description, 
+    'author': { 
+    'name': author,
+      'icon_url': authorURL
+    },
+    'image': { 
+      'url': image 
+    }, 
+    'timestamp': timestamp, 
+    'footer': { 
+      'text': footer 
+    } 
+  };
+  return embed; 
+};
+
+this.extension = async (reaction, attachment) => {
+  const imageLink = attachment.split('.');
+  const typeOfImage = imageLink[imageLink.length - 1];
+  const image = /(jpg|jpeg|png|gif)/gi.exec(typeOfImage);
+  if (!image) return '';
+  return attachment;
+};
+```
+
+This creates 2 functions attached to `this.` for later convenience. `this.starEmbed` will create the embed for us to send. Next, we create a function that will determine whether or not there is an image attached to the message.
+
+Let's keep going.
+
+```js
+async run(reaction, user) {
+  const message = reaction.message;
+  const { starboardChannel } = this.client.settings.get(message.guild.id)
```

Comment:
You're not using this yet, I'd suggest to retrieve the settings before you need them, not before non-settings related checks, and if you're using destructuring, why you aren't doing the same in the line above? Also, you missed a semicolon.